game/core/game_map.py

```diff
@@ -3,35 +3,72 @@
 from collections import OrderedDict
 
 import arcade
+from arcade.sprite import Sprite
 from arcade.tilemap import load_tilemap
+from beartype import beartype
 from loguru import logger
 
+from .game_clock import GameClock
+from .game_state import GameState
 from .views.animated_sprite import AnimatedSprite
 
 
 class GameMap:
     """Model the Game's Tile Map."""
 
-    def __init__(self, state, game_clock):  # type: ignore[no-untyped-def]
-        self.tile_map = state.map_path
-        self.load(state.inverse_movement)  # type: ignore[no-untyped-call]
+    @beartype
+    def __init__(self, state: GameState, game_clock: GameClock) -> None:
+        self.state = state
+        self.game_clock = game_clock
+        self.load()
 
         self.sparkles = arcade.SpriteList()
         for item in self.map_layers.get("searchable", []):
             self.sparkles.append(
                 AnimatedSprite(
-                    game_clock, "sparkle", item.center_x, item.center_y, item, 0.8
+                    self.game_clock,
+                    "sparkle",
+                    item.center_x,
+                    item.center_y,
+                    item,
+                    scale=0.8,
                 )
             )
 
+    @beartype
+    def remove_sprite(self, removed_sprite: Sprite, searchable: bool) -> None:
+        removed_sprite.properties["removed"] = True
+        removed_sprite.remove_from_sprite_lists()
+        if dropped_item := self.state.sync_removed_sprite(removed_sprite, searchable):
+            new_sprite = AnimatedSprite(
+                self.game_clock,
+                "sparkle",
+                dropped_item.center_x,
+                dropped_item.center_y,
+                dropped_item,
+                scale=0.8,
+            )
+            self.sparkles.append(new_sprite)
+            # FIXME: Why is the sprite not being drawn?
+            logger.warning(
+                f"Expected new sprite at ({dropped_item.center_x}, {dropped_item.center_y})"
+            )
+            self.draw()
+            new_sprite.draw()
+            dropped_item.draw()
+            self.on_update()
```

Comment:
@dimmerex, do you know why the sprite isn't being drawn?

I thought it should be just:

```suggestion
            self.draw()
```

But none of these combinations are working

Reply:
Once this is fixed, the PR is ready to merge!

Reply:
Will merge for now and fix in a future PR

```suggestion
            # FIXME: Reload the entire game map rather than trying to be too smart
            logger.warning(
                f"Expected to draw new sprite at ({dropped_item.center_x}, {dropped_item.center_y})"
            )
```